File: content/projectilewars/panorama/scripts/custom_game/hero_selection.js
"use strict";
// FUCK THIS
function SetupHeroSelection() {
    var hudElements = $.GetContextPanel().GetParent().GetParent().FindChildTraverse("HUDElements");
    var dotaHud = $.GetContextPanel().GetParent().GetParent().GetParent();
    var customUI = dotaHud.FindChildTraverse("PreGame");
    var heroSelection;
    if (heroSelection == null) {
        heroSelection = $.CreatePanel("Panel", customUI, "HeroSelection");
        heroSelection.BLoadLayout("file://{resources}/layout/custom_game/hero_selection.xml", false, false);
        heroSelection.style.visibility = "visible";
    }
}
function MarkUnavailableHeroes(kv) {
}
function SubmitHeroPick(heroname) {
    GameEvents.SendCustomGameEventToServer("submit_hero_pick", { heroName: heroname });
}
GameEvents.Subscribe("mark_unavailable_heroes", MarkUnavailableHeroes);

File: content/projectilewars/panorama/scripts/custom_game/hero_selection.lua
require("typescript_lualib")
function SetupHeroSelection()
    local hudElements = Panel.FindChildTraverse(Panel.GetParent(Panel.GetParent(DollarStatic.GetContextPanel($))),"HUDElements")

    local dotaHud = Panel.GetParent(Panel.GetParent(Panel.GetParent(DollarStatic.GetContextPanel($))))

    local customUI = Panel.FindChildTraverse(dotaHud,"PreGame")

    local heroSelection = nil

    if heroSelection==nil then
        heroSelection=DollarStatic.CreatePanel($,"Panel",customUI,"HeroSelection")
        heroSelection.BLoadLayout(heroSelection,"file://{resources}/layout/custom_game/hero_selection.xml",false,false)
        heroSelection.style.visibility="visible"
    end
end
function MarkUnavailableHeroes(kv)
end
function SubmitHeroPick(heroname)
    CDOTA_PanoramaScript_GameEvents.SendCustomGameEventToServer(GameEvents,"submit_hero_pick",{heroName=heroname})
end
CDOTA_PanoramaScript_GameEvents.Subscribe(GameEvents,"mark_unavailable_heroes",MarkUnavailableHeroes)
